components/automate-cli/cmd/chef-automate/pullAndGenerateConfig.go

```diff
@@ -385,23 +422,27 @@ func determineDBType(a2ConfigMap map[string]*dc.AutomateConfig, dbtype string) (
 	return "", errors.New(`unsupported db type. It should be either "aws" or "self-managed" or ""`)
 }
 
-func (p *PullConfigsImpl) fetchInfraConfig() (*ExistingInfraConfigToml, error) {
+func (p *PullConfigsImpl) fetchInfraConfig(removeUnreachableNodes bool) (*ExistingInfraConfigToml, map[string][]string, error) {
+	unreachableNodes := make(map[string][]string)
 	sharedConfigToml, err := getExistingHAConfig()
 	if err != nil {
-		return nil, status.Wrap(err, status.ConfigError, "unable to fetch HA config")
+		return nil, nil, status.Wrap(err, status.ConfigError, "unable to fetch HA config")
 	}
-	a2ConfigMap, err := p.pullAutomateConfigs()
+	a2ConfigMap, automateUnreachableNodes, err := p.pullAutomateConfigs(removeUnreachableNodes)
 	if err != nil {
-		return nil, status.Wrap(err, status.ConfigError, "unable to fetch Automate config")
+		return nil, nil, status.Wrap(err, status.ConfigError, "unable to fetch Automate config")
 	}
-	csConfigMap, err := p.pullChefServerConfigs()
+	unreachableNodes[AUTOMATE] = automateUnreachableNodes
+
+	csConfigMap, chefServerUnreachableNodes, err := p.pullChefServerConfigs(removeUnreachableNodes)
```

Comment:
**Scenario: Nodes are not reachable**
When the nodes are not reachable, I think this would take a significant time to respond with an error. Can we make the following call as a go routine so that the process wouldn't block the subsequent calls in the for the other services, 
**Calls:**
- pullAutomateConfigs
- pullChefServerConfigs
- pullPGConfigs
- pullOpensearchConfigs
Let's discuss if that would be a feasible change and would reduce the process time with the team and then we can implement based on the discussion

Reply:
this functions are used so many places like cert-rotate, config show, upgrade etc, so making it async will need significant amount of testing of all the flow